src/zenml/pipelines/base_pipeline.py

```diff
@@ -158,7 +158,7 @@ def name(self) -> str:
         return self.configuration.name
 
     @property
-    def enable_cache(self) -> bool:
+    def enable_cache(self) -> Optional[bool]:
```

Comment:
Actually now looking at this again, can't we make this non-optional still? I mean, not specifying anything in the config is essentially the same as having the cache enabled right?

Reply:
Same for `BaseStep` and post execution `PipelineRun`

Reply:
I would suggest to leave it as is. The current properties are all called `enable_cache`, not `cache_enabled` or similar, so I would personally expect that the property simply reflects the setting of the user.